### ISSUE TITLE
readSTL should check file header

### ISSUE THREAD

**Issue by medakk** [libigl/libigl#1738]
#### Describe the bug
I accidentally called readSTL on a .ply file. Totally my fault, but it'd be nice if the readSTL function had checked the header and made sure it was a STL file. Instead, it just ate up all the memory and crashed with a SIGKILL after OOMing.

#### Platform

- [ ] Windows
- [ ] macOS
- [x] Linux

**Comment by jdumas** [comment]
You can use the `igl::read_triangle_mesh` instead. `readSTL` shouldn't check the file extension, but admittedly it shouldn't crash either. Could you share the ply you tried it with and we can see if we can fix the issue and produce an error message early on?

**Comment by medakk** [comment]
I created a minimal 1-face mesh that reproduces the problem:

[crash_test.zip](https://github.com/libigl/libigl/files/6058176/crash_test.zip)

**Comment by medakk** [comment]
> readSTL shouldn't check the file extension

I agree, I meant that it should check the header to determine if its a STL file. But I just realized that that is [probably impossible because a binary STL file doesn't specify a "magic string"](https://en.wikipedia.org/wiki/STL_(file_format)#Binary_STL)
